lib/assets/javascripts/turbolinks.coffee

```diff
@@ -142,15 +142,20 @@ constrainPageCacheTo = (limit) ->
 
 replace = (html, options = {}) ->
   loadedNodes = changePage extractTitleAndBody(createDocument(html))..., options
-  triggerEvent (if options.change then EVENTS.PARTIAL_LOAD else EVENTS.LOAD), loadedNodes
+  triggerEvent (if options.change or options.append or options.prepend then EVENTS.PARTIAL_LOAD else EVENTS.LOAD), loadedNodes
 
 changePage = (title, body, csrfToken, options) ->
   title = options.title ? title
   currentBody = document.body
 
-  if options.change
-    nodesToChange = findNodes(currentBody, '[data-turbolinks-temporary]')
-    nodesToChange.push(findNodesMatchingKeys(currentBody, options.change)...)
+  if options.change or options.append or options.prepend
+    nodesToAppend = findNodesMatchingKeys(currentBody, options.append) if options.append
```

Comment:
I don't understand.

These are separated out because they are treated differently during replacement (the nodesToAppend are passed to swapNodes for appending and become appendedNodes, etc.).

See L167 - L169.